tests/conftest.py

```diff
@@ -38,7 +38,7 @@ def cert_and_key(tmpdir):
 
 @pytest.fixture
 def sp_conf(cert_and_key):
-    sp_base = "http://example.com"
+    sp_base = "http://example.com/test"
```

Comment:
we should parametrize the possible values for the sp and idp and have pytest run the combos, so that we test all cases.